sw/airborne/modules/loggers/sdlog_chibios.c

```diff
@@ -106,6 +106,20 @@ static enum {
 static char chibios_sdlog_filenames[68];
 static char NO_FILE_NAME[] = "none";
 
+#if PREFLIGHT_CHECKS
+/* Preflight checks */
+#include "modules/checks/preflight_checks.h"
+static struct preflight_check_t sdlog_pfc;
+
+static void sdlog_preflight(struct preflight_result_t *result) {
+  if(chibios_sdlog_status != SDLOG_RUNNING) {
```

Comment:
I suggest to set to error only if status is `SDLOG_ERROR`. With the startup delay, it can block you for tuning on the motors. If you don't have an SD card, I'm not even sure what is happening, but your checks will always fail then.
An other option is to have a "warning" check. Something that reports a message but doesn't block the fail check ?

Reply:
We currently prefer to not takeoff when we are not logging, but I can make this check optional. Also could add a warning option, such that you still get a message in the GCS.
Since most of our platforms are research platforms you don't want to take off if the SDLogger is not running for some reason, even with the startup delay you just have to wait. Since most tests/flights are useless if something goes wrong if you don't have an onboard log. If you still want to take off without logging you can use the setting to bypass the checks, but this would require a user decision and active action.

But to make this work for most people I'm also happy to add some of these checks behind defines/warnings by default, such that you can choose yourself what you prefer.

Reply:
If you don't have an SD card the logger status goes to `ERROR`, but also then we prefer not to take off. Since most of the time this is a user error forgetting to place the SD card back in the UAV. If you don't want to fly with the logger you should just remove the module from your airframe, or bypass the checks(if only temporarily).

Reply:
This now changed to a warning by default